libtenzir/builtins/formats/xsv.cpp

```diff
@@ -185,41 +250,41 @@ auto parse_impl(generator<std::optional<std::string_view>> lines,
   }
   if (!header || header->empty())
     co_return;
-  auto split_parser
-    = (((parsers::qqstr
-           .then([](std::string in) {
-             static auto unescaper = [](auto& f, auto l, auto out) {
-               if (*f != '\\') { // Skip every non-escape character.
-                 *out++ = *f++;
-                 return true;
-               }
-               if (l - f < 2)
-                 return false;
-               switch (auto c = *++f) {
-                 case '\\':
-                   *out++ = '\\';
-                   break;
-                 case '"':
-                   *out++ = '"';
-                   break;
-               }
-               ++f;
-               return true;
-             };
-             return detail::unescape(in, unescaper);
-           })
-           .with([](const std::string& in) {
-             return !in.empty();
-           })
-         >> &(sep | parsers::eoi))
-        | +(parsers::any - sep))
-       % sep);
+  auto string_value_parser
+    = ((parsers::qqstr
+          .then([](std::string in) {
+            static auto unescaper = [](auto& f, auto l, auto out) {
+              if (*f != '\\') { // Skip every non-escape character.
+                *out++ = *f++;
+                return true;
+              }
+              if (l - f < 2)
+                return false;
+              switch (auto c = *++f) {
+                case '\\':
+                  *out++ = '\\';
+                  break;
+                case '"':
+                  *out++ = '"';
+                  break;
+              }
+              ++f;
+              return true;
+            };
+            return detail::unescape(in, unescaper);
+          })
+          .with([](const std::string& in) {
+            return !in.empty();
+          })
```

Comment:
I know that this check is not new, but I don't think it makes much sense. If we parse quoted strings, then `""` should also be allowed, otherwise `""` gets parsed as `"\"\""`.

Reply:
I rebased after merging https://github.com/tenzir/tenzir/pull/3681 and have made this change as well.